### ISSUE TITLE
Error on validate with active_url rule

### ISSUE THREAD

**Issue by m33ch** [laravel/framework#16591]
- Laravel Version: 5.2.45
- PHP Version: 5.6.21
- Database Driver & Version: ND

### Description:
Error on validate url with `active_url` rule.

### Steps To Reproduce:
I set `active_url` rule on field `url` and when i try to post data, server response: 

```
exception 'ErrorException' with message 'dns_get_record(): A temporary server error occurred.' in /var/www/console/vendor/laravel/framework/src/Illuminate/Validation/Validator.php:1554
Stack trace:
#0 [internal function]: Illuminate\Foundation\Bootstrap\HandleExceptions->handleError(2, 'dns_get_record(...', '/var/www/console...', 1554, Array)
#1 /var/www/console/vendor/laravel/framework/src/Illuminate/Validation/Validator.php(1554): dns_get_record('www.prova.it', 134217729)
#2 /var/www/console/vendor/laravel/framework/src/Illuminate/Validation/Validator.php(485): Illuminate\Validation\Validator->validateActiveUrl('callback_url', 'http://www.prov...', Array, Object(Illuminate\Validation\Validator))
#3 /var/www/console/vendor/laravel/framework/src/Illuminate/Validation/Validator.php(425): Illuminate\Validation\Validator->validate('callback_url', 'active_url')
#4 /var/www/console/vendor/laravel/framework/src/Illuminate/Validation/Validator.php(450): Illuminate\Validation\Validator->passes()
#5 /var/www/console/vendor/prettus/laravel-validation/src/Prettus/Validator/LaravelValidator.php(39): Illuminate\Validation\Validator->fails()
#6 /var/www/console/vendor/prettus/laravel-validation/src/Prettus/Validator/AbstractValidator.php(108): Prettus\Validator\LaravelValidator->passes('create')
#7 /var/www/console/app/Services/Batch/Creator.php(45): Prettus\Validator\AbstractValidator->passesOrFail('create')
#8 /var/www/console/app/Services/Batch/Cruder.php(30): App\Services\Batch\Creator->store()
#9 /var/www/console/app/Http/Controllers/Api/v1/BatchesController.php(159): App\Services\Batch\Cruder->createByRequest(Object(Dingo\Api\Http\Request))
#10 [internal function]: App\Http\Controllers\Api\v1\BatchesController->store(Object(Dingo\Api\Http\Request))
#11 /var/www/console/bootstrap/cache/compiled.php(9424): call_user_func_array(Array, Array)
#12 /var/www/console/bootstrap/cache/compiled.php(9486): Illuminate\Routing\Controller->callAction('store', Array)
#13 /var/www/console/bootstrap/cache/compiled.php(9466): Illuminate\Routing\ControllerDispatcher->call(Object(App\Http\Controllers\Api\v1\BatchesController), Object(Illuminate\Routing\Route), 'store')
#14 [internal function]: Illuminate\Routing\ControllerDispatcher->Illuminate\Routing\{closure}(Object(Dingo\Api\Http\Request))
#15 /var/www/console/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(52): call_user_func(Object(Closure), Object(Dingo\Api\Http\Request))
#16 [internal function]: Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Dingo\Api\Http\Request))
#17 /var/www/console/bootstrap/cache/compiled.php(9948): call_user_func(Object(Closure), Object(Dingo\Api\Http\Request))
#18 /var/www/console/bootstrap/cache/compiled.php(9467): Illuminate\Pipeline\Pipeline->then(Object(Closure))
#19 /var/www/console/bootstrap/cache/compiled.php(9454): Illuminate\Routing\ControllerDispatcher->callWithinStack(Object(App\Http\Controllers\Api\v1\BatchesController), Object(Illuminate\Routing\Route), Object(Dingo\Api\Http\Request), 'store')
#20 /var/www/console/bootstrap/cache/compiled.php(8524): Illuminate\Routing\ControllerDispatcher->dispatch(Object(Illuminate\Routing\Route), Object(Dingo\Api\Http\Request), 'App\\Http\\Co...', 'store')
#21 /var/www/console/bootstrap/cache/compiled.php(8511): Illuminate\Routing\Route->runController(Object(Dingo\Api\Http\Request))
#22 /var/www/console/bootstrap/cache/compiled.php(8225): Illuminate\Routing\Route->run(Object(Dingo\Api\Http\Request))
#23 [internal function]: Illuminate\Routing\Router->Illuminate\Routing\{closure}(Object(Dingo\Api\Http\Request))
#24 /var/www/console/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(52): call_user_func(Object(Closure), Object(Dingo\Api\Http\Request))
#25 /var/www/console/vendor/dingo/api/src/Http/Middleware/Auth.php(55): Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Dingo\Api\Http\Request))
#26 [internal function]: Dingo\Api\Http\Middleware\Auth->handle(Object(Dingo\Api\Http\Request), Object(Closure))
#27 /var/www/console/bootstrap/cache/compiled.php(9963): call_user_func_array(Array, Array)
#28 [internal function]: Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Dingo\Api\Http\Request))
#29 /var/www/console/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(32): call_user_func(Object(Closure), Object(Dingo\Api\Http\Request))
#30 /var/www/console/vendor/dingo/api/src/Http/Middleware/PrepareController.php(45): Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Dingo\Api\Http\Request))
#31 [internal function]: Dingo\Api\Http\Middleware\PrepareController->handle(Object(Dingo\Api\Http\Request), Object(Closure))
#32 /var/www/console/bootstrap/cache/compiled.php(9963): call_user_func_array(Array, Array)
#33 [internal function]: Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Dingo\Api\Http\Request))
#34 /var/www/console/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(32): call_user_func(Object(Closure), Object(Dingo\Api\Http\Request))
#35 [internal function]: Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Dingo\Api\Http\Request))
#36 /var/www/console/bootstrap/cache/compiled.php(9948): call_user_func(Object(Closure), Object(Dingo\Api\Http\Request))
#37 /var/www/console/bootstrap/cache/compiled.php(8226): Illuminate\Pipeline\Pipeline->then(Object(Closure))
#38 /var/www/console/bootstrap/cache/compiled.php(8217): Illuminate\Routing\Router->runRouteWithinStack(Object(Illuminate\Routing\Route), Object(Dingo\Api\Http\Request))
#39 /var/www/console/bootstrap/cache/compiled.php(8207): Illuminate\Routing\Router->dispatchToRoute(Object(Dingo\Api\Http\Request))
#40 /var/www/console/vendor/dingo/api/src/Routing/Adapter/Laravel.php(81): Illuminate\Routing\Router->dispatch(Object(Dingo\Api\Http\Request))
#41 /var/www/console/vendor/dingo/api/src/Routing/Router.php(513): Dingo\Api\Routing\Adapter\Laravel->dispatch(Object(Dingo\Api\Http\Request), 'v1')
#42 /var/www/console/vendor/dingo/api/src/Http/Middleware/Request.php(126): Dingo\Api\Routing\Router->dispatch(Object(Dingo\Api\Http\Request))
#43 [internal function]: Dingo\Api\Http\Middleware\Request->Dingo\Api\Http\Middleware\{closure}(Object(Dingo\Api\Http\Request))
#44 /var/www/console/bootstrap/cache/compiled.php(9970): call_user_func(Object(Closure), Object(Dingo\Api\Http\Request))
#45 /var/www/console/app/Http/Middleware/JsonRequestValidator.php(28): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Dingo\Api\Http\Request))
#46 [internal function]: App\Http\Middleware\JsonRequestValidator->handle(Object(Dingo\Api\Http\Request), Object(Closure))
#47 /var/www/console/bootstrap/cache/compiled.php(9963): call_user_func_array(Array, Array)
#48 /var/www/console/bootstrap/cache/compiled.php(3286): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Dingo\Api\Http\Request))
#49 [internal function]: Illuminate\Foundation\Http\Middleware\CheckForMaintenanceMode->handle(Object(Dingo\Api\Http\Request), Object(Closure))
#50 /var/www/console/bootstrap/cache/compiled.php(9963): call_user_func_array(Array, Array)
#51 [internal function]: Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Dingo\Api\Http\Request))
#52 /var/www/console/bootstrap/cache/compiled.php(9948): call_user_func(Object(Closure), Object(Dingo\Api\Http\Request))
#53 /var/www/console/vendor/dingo/api/src/Http/Middleware/Request.php(127): Illuminate\Pipeline\Pipeline->then(Object(Closure))
#54 /var/www/console/vendor/dingo/api/src/Http/Middleware/Request.php(103): Dingo\Api\Http\Middleware\Request->sendRequestThroughRouter(Object(Dingo\Api\Http\Request))
#55 [internal function]: Dingo\Api\Http\Middleware\Request->handle(Object(Illuminate\Http\Request), Object(Closure))
#56 /var/www/console/bootstrap/cache/compiled.php(9963): call_user_func_array(Array, Array)
#57 [internal function]: Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#58 /var/www/console/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(32): call_user_func(Object(Closure), Object(Illuminate\Http\Request))
#59 [internal function]: Illuminate\Routing\Pipeline->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#60 /var/www/console/bootstrap/cache/compiled.php(9948): call_user_func(Object(Closure), Object(Illuminate\Http\Request))
#61 /var/www/console/bootstrap/cache/compiled.php(2366): Illuminate\Pipeline\Pipeline->then(Object(Closure))
#62 /var/www/console/bootstrap/cache/compiled.php(2350): Illuminate\Foundation\Http\Kernel->sendRequestThroughRouter(Object(Illuminate\Http\Request))
#63 /var/www/console/public/index.php(54): Illuminate\Foundation\Http\Kernel->handle(Object(Illuminate\Http\Request))
#64 {main}
```

**Comment by fernandobandeira** [comment]
Are you running this on homestead perhaps?

**Comment by m33ch** [comment]
No it's on production server.

**Comment by fernandobandeira** [comment]
Check the output of

```php
$result = dns_get_record("php.net");
print_r($result);
```

This is usually a dns issue, it should be returning false if that's the reason...

**Comment by m33ch** [comment]
I tried to execute in shell and it's work!
My url passed via post is "prova.it". 
I tried in shell with my parameter and return: 

```
PHP Warning:  dns_get_record(): A temporary server error occurred. in php shell code on line 1
```

**Comment by m33ch** [comment]
Why Laravel catch `ErrorException` instead to catch `ValidatorException` ?

Thanks

**Comment by fernandobandeira** [comment]
This was fixed on 5.3, by the looks of it

To fix your issue, create a custom validator:

https://laravel.com/docs/5.2/validation#custom-validation-rules

And put this code inside it:
```php
        if (! is_string($value)) {
            return false;
        }
        if ($url = parse_url($value, PHP_URL_HOST)) {
            try {
                return count(dns_get_record($url, DNS_A | DNS_AAAA)) > 0;
            } catch (Exception $e) {
                return false;
            }
        }
        return false;
```

And this issue should be closed since 5.2 isn't supported anymore...

**Comment by m33ch** [comment]
Ok, thank you very much!
